### ISSUE TITLE
remove inline flag from FunctionSignature

### ISSUE THREAD

**Issue by coolya** [mbeddr/mbeddr.core#1267]
The inline flag should be removed from the FunctionSignature and replaced with a InlinePrefix. Platform specific generators can then map this prefix to the appropriated inline keyword for the current compiler. The generic InlinePrefix shall not have a textgen so generation fails if the platform specific compiler does not map it to something useful.

**Comment by stephaneberle9** [comment]
I disagree with this idea. "inline" is not a complier-specific "flag" but a true keyword that is part of the C (and C++) language (see e.g., https://en.wikipedia.org/wiki/Inline_function). The thing that varies from compiler to compiler is not the syntax used to inline a function but its semantics, i.e., the way how an inlined function translates to machine code.
